### ISSUE TITLE
Empty init() being generated for a MutationInput struct

### ISSUE THREAD

**Issue by arvindravi** [apollostack/apollo-codegen#22]
Hi there,

`apollo-codegen` seems to generate an empty `init()` on a mutation struct in `API.swift`, my server implementation for the mutation is pretty straightforward with one input object (`UserInput`) and one return type (`UserType`)

Xcode won't compile with an empty init because the `graphQLMap` property is not being initialised. The workaround currently is modifying `API.swift` and initialising `graphQLMap` inside the empty init before running the project, like:

```swift
public init() {
  graphQLMap = [:]
}
```

Steps to reproduce:

1. Define a Mutation on the server that takes an input object
2. Use `apollo-codegen` to download the schema and copy it to your target
3. Build your project
4. The generated `API.swift` has an empty `public init()`

Can anyone else reproduce this?

This is my API.swift for reference: [API.swift](https://gist.github.com/arvindravi/d5ea8609f00ca7e6ba2b5912c06dca81)

Thanks for the wonderful set of tools, otherwise. :)

**Comment by martijnwalraven** [comment]
@arvindravi: Thanks for reporting this. It should be fixed in `apollo-codegen` 0.9.6.

The issue is triggered when an input type only contains optional fields, because otherwise an empty constructor will not be generated. While this is valid GraphQL, in most cases at least some of the values should be required because it doesn't make sense to pass in an empty input object.

So in your case, you may want to change your type definition to:

```graphql
input CreateUserMutationInput {
  userInput: UserInput!
  clientMutationId: String!
}
```

**Comment by arvindravi** [comment]
Thanks for the fix and pointing out the issue, Martijn!
